strongbox-data-service/src/main/java/org/carlspring/strongbox/data/criteria/OQueryTemplate.java

```diff
@@ -0,0 +1,209 @@
+                                                   int i)
+    {
+        HashMap<String, Object> result = new HashMap<>();
+        Expression e = p.getExpression();
+        if (e != null)
+        {
+            result.put(calculateParameterName(e.getProperty(), i), e.getValue());
+        }
+
+        for (Predicate predicate : p.getChildPredicateList())
+        {
+            result.putAll(exposeParameterMap(predicate, i++));
+        }
+
+        return result;
+    }
+
+    public String calculateQueryString(Selector<T> selector,
+                                       Paginator predicate)
+    {
+        StringBuilder sb = new StringBuilder();
+        sb.append("SELECT ").append(selector.getProjection());
+
+        sb.append(" FROM ").append(selector.getTargetClass().getSimpleName());
+
+        Predicate p = selector.getPredicate();
+        if (p.getChildPredicateList().isEmpty() && p.getExpression() == null)
+        {
+            return sb.toString();
+        }
+
+        sb.append(" WHERE ");
+        sb.append(predicateToken(p, 0));
+
+        if (predicate != null && predicate.getOrderBy() != null && !predicate.getOrderBy().trim().isEmpty())
+        {
+            sb.append(String.format(" ORDER BY %s", predicate.getOrderBy()));
+        }
+        
+        if (predicate != null && predicate.getSkip() > 0)
+        {
+            sb.append(String.format(" SKIP %s", predicate.getSkip()));
+        }
+        if (predicate != null && predicate.getLimit() > 0)
+        {
+            sb.append(String.format(" LIMIT %s", predicate.getLimit()));
+        }
+
+        if (selector.isFetch())
+        {
+            sb.append(" FETCHPLAN *:-1");
+        }
+
+        return sb.toString();
+    }
+
+    protected String predicateToken(Predicate p,
+                                    int i)
+    {
+        StringBuffer sb = new StringBuffer();
+        if (p.getExpression() != null)
+        {
+            sb.append(expressionToken(p.getExpression(), i));
+        }
+
+        for (Predicate predicate : p.getChildPredicateList())
+        {
+            if (sb.length() > 0)
+            {
+                sb.append(String.format(" %s ", p.getOperator().name()));
+            }
+            sb.append(predicateToken(predicate, i++));
+        }
+
+        return sb.toString();
+    }
+
+    protected String expressionToken(Expression e,
+                                     int n)
+    {
+        String experssionLeft = expressionLeftToken(e);
+        String operator = expressionOperatorToken(e);
+        String expressionRight = expressionRightToken(e, n);
+
+        return new StringBuffer().append(experssionLeft)
+                                 .append(operator)
+                                 .append(expressionRight)
+                                 .toString();
+    }
+
+    protected String expressionLeftToken(Expression e)
+    {
+        switch (e.getOperator())
+        {
```

Comment:
A-ha! :)
So, it was you! (Haha! I've seen these blocks formatted like this and I was wondering who it was, but never checked)...

Would you mind indenting these like this, please?
```
        switch (e.getOperator())
        {
            case CONTAINS:
                String property = e.getProperty();
                return property.substring(0, property.indexOf("."));
            default:
                break;
        }
```

Thanks! :)

Reply:
yeah
this is Eclipse :)
done

Reply:
Thanks! :)